### PR TITLE
fix publish for stable branch

### PR DESCRIPTION
**Pull request by mansona**
we [needed to update relase-plan](https://github.com/embroider-build/release-plan/pull/38) to add an option `--publish-branch` because by default pnpm only allows you to run `pnpm publish` on [main or master](https://pnpm.io/cli/publish#--publish-branch-branch). We are running it on `stable` and it was hanging in CI because it was prompting to know if you wanted to continue.

This PR prevents that from happening again 👍

### CODE REVIEW

File: .github/workflows/publish-stable.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
           registry-url: 'https://registry.npmjs.org'
 
       - name: npm publish
-        run: pnpm release-plan publish
+        run: pnpm release-plan publish --publish-branch=stable
```
Comment (NullVoxPopuli) [inline .github/workflows/publish-stable.yml:52]: don' we want `--publish-branch=stable|main`, since we merge `stable` into `main` frequently?

Comment (mansona) [reply .github/workflows/publish-stable.yml:52]: once we sync this it won't be a problem any more. For example https://github.com/embroider-build/embroider/pull/1721 didn't even try to update the yaml branches

also it's better to be explicit with this. 

and also we didn't test if that syntax works (if the pnpm docs says it does that's fine but I'm extra cautious around releasing things). We know what this PR has does work because I've done it manually twice.

So in summary, no i don't think we should change this 😂

Comment (NullVoxPopuli) [reply .github/workflows/publish-stable.yml:52]: ah ok, I see [what you mean](https://github.com/embroider-build/embroider/pull/1724/files/fbffd7a761762f6da8683dfb235e2acac457d7ee#diff-ee4f3ba7c21547934a6f659fce2ab1c2a42251b904c6844fad629c9f28a991aeR11)

Legit

**Review by NullVoxPopuli — APPROVED** [review]
